filters/auth/oidc.go

```diff
@@ -1075,7 +1075,9 @@ func (f *tokenOidcFilter) resolveDistributedClaimAzure(url *url.URL, oauth2Token
 		return nil, fmt.Errorf("unabled to decode response: %w", err)
 	}
 	for _, v := range target.Value {
-		values = append(values, v.DisplayName)
+		if v.OnPremisesSamAccountName != "" {
```

Comment:
> Switching to the guaranteed unique identifier onPremisesSamAccountName available in the Microsoft Graph API

Could it be empty?

Reply:
And should we have a test to ensure behaviour when it is empty?

Reply:
Yes it can be empty and extended a test to reflect it.

Reply:
`onPremisesSamAccountName` does not look like a standard attribute so we have to rely on you testing that this works correctly.

I guess `id` is also not stable/reliable to use for the purpose, is it?

Reply:
The onPremisesSamAccountName is the equivalent of the samAccountName issued with the standard group claim. 

But it might be that someone rather would like the GUID / ID to be issued, it is configurable in Azure app registration. 

If you want @AlexanderYastrebov the field selector could be extended to be configurable. If so, which way, using a flag or another filter parameter?

Reply:
We do not use azure-based auth and have no experience with it. Lets have it like that and see if we get any feedback on the change from other users.